process/block/preprocess/basePreProcess.go

```diff
@@ -171,35 +174,38 @@ func (bpp *basePreProcess) computeExistingAndMissing(
 	missingTxsForShard := make(map[uint32][]*txsHashesInfo, 0)
 
 	forBlock.mutTxsForBlock.Lock()
-	for i := 0; i < len(body); i++ {
-		miniBlock := body[i]
-		if miniBlock.Type != currType {
-			continue
-		}
-
-		txShardInfo := &txShardInfo{senderShardID: miniBlock.SenderShardID, receiverShardID: miniBlock.ReceiverShardID}
-		txHashes := make([][]byte, 0)
-
-		for j := 0; j < len(miniBlock.TxHashes); j++ {
-			txHash := miniBlock.TxHashes[j]
-			tx, err := process.GetTransactionHandlerFromPool(
-				miniBlock.SenderShardID,
-				miniBlock.ReceiverShardID,
-				txHash,
-				txPool)
 
-			if err != nil {
-				txHashes = append(txHashes, txHash)
-				forBlock.missingTxs++
+	if body != nil {
```

Comment:
please use this construction instead:

if body == nil {
return process.ErrNilBlockBody
}


for....

Reply:
done